### PR TITLE
[Console][Table] Add support for colspan/rowspan + multiple header lines

### PR DESCRIPTION
**Pull request by aitboudad**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | #13368, #13369 |
| Tests pass? | yes |
| License | MIT |

This PR introduce a new feature described in #13368 and #13369,
I created a new class `TableCell` which can allow us to specify colspan/rowspan for each cell.

``` php
$table->addRow([new TableCell("data", array('rowspan' => 1, 'colspan' => 2)), 'data']);
```
- [x] #13368 Add support for colspan/rowspan
- [x] #13369 Add support for multiple header lines
- [ ] add doc

### CODE REVIEW

File: src/Symfony/Component/Console/Helper/TableCell.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        return $this->value;
+    }
+
+    /**
+     * Sets the cell value.
+     *
+     * @return string
+     */
+    public function setValue($value)
```
Comment (unkind) [inline src/Symfony/Component/Console/Helper/TableCell.php]: I'd suggest to avoid this setter as `TableCell` is a value object.

Comment (stof) [inline src/Symfony/Component/Console/Helper/TableCell.php]: agreed. It should be immutable

Comment (aitboudad) [inline src/Symfony/Component/Console/Helper/TableCell.php]: fixed :)

**Comment by aitboudad** [comment]
This PR is now ready to be reviewed.

**Comment by jeremy-derusse** [comment]
Tests missed a case with a separator inside a rowspan

```
+------------------+-----------------+
| ISBN             | Author          |
+------------------+-----------------+
| 9971-5-0210-0    | Dante Alighieri |
|                  +-----------------+
|                  | Charles Dickens |
+------------------+-----------------+
```

**Comment by aitboudad** [comment]
@jeremy-derusse done :)

**Comment by aitboudad** [comment]
ping @symfony/deciders

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php [hunk 2]
```diff
@@ -252,6 +253,214 @@ public function testRenderProvider()
...
+| ISBN           | Title         | Author          |
++----------------+---------------+-----------------+
+| 99921-58-10-7  | Divine Comedy | Dante Alighieri |
++----------------+---------------+-----------------+
+| Divine Comedy(Dante Alighieri)                   |
++----------------+---------------+-----------------+
+| Arduino: A Quick-Start Guide   | Mark Schmidt    |
++----------------+---------------+-----------------+
+| 9971-5-0210-0  | A Tale of                       |
+|                | Two Cities                      |
```
Comment (fabpot) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]: I would expect:

```
+----------------+---------------+-----------------+
| 9971-5-0210-0  | A Tale of Two Cities            |
+----------------+---------------+-----------------+
```

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php [hunk 3]
```diff
@@ -252,6 +253,214 @@ public function testRenderProvider()
...
+| ISBN          | Title                | Author          |
++---------------+----------------------+-----------------+
+| 9971-5-0210-0 | Divine Comedy        | Dante Alighieri |
+|               | A Tale of Two Cities | Charles Dickens |
+|               | The Lord of          | J. R.           |
+|               | the Rings            | R. Tolkien      |
++---------------+----------------------+-----------------+
+| 80-902734-1-6 | And Then             | Agatha Christie |
+| 80-902734-1-7 | There                | Test            |
+|               | Were None            |                 |
```
Comment (fabpot) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]: Same here.

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php [hunk 4]
```diff
@@ -252,6 +253,214 @@ public function testRenderProvider()
...
+| 9971                    | Dante Alighieri |
+| -5-                     | Charles Dickens |
+| 021                     |                 |
+| 0-0                     |                 |
++-----------------+-------+-----------------+
+| Dante Alighieri | 9971                    |
+| Charles Dickens | -5-                     |
+|                 | 021                     |
+|                 | 0-0                     |
++-----------------+-------+-----------------+
```
Comment (fabpot) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]: I would expect:

```
+-----------------+-------+-----------------+
| Dante Alighieri | 9971                    |
| Charles Dickens | -5-0210-0               |
+-----------------+-------+-----------------+
```

You get the point. The text should take the maximum space available in the merged cells. Does it make sense?

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php (hunk 2 above)
Comment (aitboudad) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]: but cell contain new line break "A Tale of \nTwo Cities" seehttps://github.com/aitboudad/symfony/blob/console_table/src/Symfony/Component/Console/Tests/Helper/TableTest.php#L270

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php (hunk 3 above)
Comment (aitboudad) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]: same as above "And Then \nThere \nWere None"

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php (hunk 4 above)
Comment (aitboudad) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]: same here text is "9971\n-5-\n021\n0-0"

**Comment by fabpot** [comment]
@aitboudad Can you start working on the documentation?
ping @symfony/deciders

:+1:

**Comment by aitboudad** [comment]
I'll work on doc next week, so I opened an issue https://github.com/symfony/symfony-docs/issues/5119.

**Comment by fabpot** [comment]
Thank you @aitboudad.

**Pull request closed without merging**